File: metalanguage/src/main/java/constructs/MLConstruct.java
package constructs;

import lombok.Getter;
import lombok.NonNull;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

import java.util.UUID;

@Getter
@RequiredArgsConstructor
public class MLConstruct {
    @NonNull
    @Setter
    private String name;
    @NonNull
    private UUID id;
    @NonNull
    private UUID prototypeId;
    @Setter
    private UUID innerStructure;
}

File: metalanguage-lib/src/main/kotlin/ru/psu/constructs/MLConstruct.kt
package ru.psu.constructs

import java.io.Serializable
import java.util.*

//Класс конструкции MetaLanguage 1.1
abstract class MLConstruct(
        var parentId:UUID?, //Идентификатор родителя
        val id: UUID, //Идентификатор
        var name:String, //Название
        val prototypeId:UUID?, //Прототип (если null, то это конструкция в метамодели)
        val innerStructure:UUID?, //Id графа, описывающего внутреннюю структуру (если есть)
        var maxCount:Int //Максимальное число экземпляров
): Serializable